mdal/frmts/mdal_hdf5.hpp

```diff
@@ -75,16 +76,21 @@ class HdfFile
     inline std::vector<std::string> groups() const;
 
     inline HdfGroup group( const std::string &path ) const;
+    inline HdfGroup createGroup( const std::string &path ) const;
     inline HdfDataset dataset( const std::string &path ) const;
+    inline HdfDataset dataset( const std::string &path, HdfDataType dtype, size_t nItems = 1 ) const;
+    inline HdfDataset dataset( const std::string &path, HdfDataType dtype, HdfDataspace dataspace ) const;
     inline HdfAttribute attribute( const std::string &attr_name ) const;
     inline bool pathExists( const std::string &path ) const;
     std::string filePath() const;
 
   protected:
-    std::shared_ptr<Handle> d;
+    SharedHandle d;
     std::string mPath;
 };
 
+typedef std::shared_ptr<HdfFile> HdfFileShared;
```

Comment:
do we need new `typedef` personally for me it is easier to read the code when shared_ptrs are expanded everywhere.